test/Microsoft.Azure.WebJobs.Host.UnitTests/JobHostConfigurationTests.cs

```diff
@@ -325,5 +326,52 @@ private static void TestHostIdThrows(string hostId)
         private class CustomStorageClientFactory : StorageClientFactory
         {
         }
+
+
+        // Test that we can explicitly disable storage and call through a function
+        [Fact]
+        public void JobHost_NoStorage_Succeeds()
+        {
+            string prevStorage  = Environment.GetEnvironmentVariable("AzureWebJobsStorage");
+            string prevDashboard = Environment.GetEnvironmentVariable("AzureWebJobsDashboard");
+            try
+            {
+                Environment.SetEnvironmentVariable("AzureWebJobsStorage", null);
+                Environment.SetEnvironmentVariable("AzureWebJobsDashboard", null);
```

Comment:
Do you need to modify machine settings like this if you're setting explicitly below? If this test fails mid way and the finally block isn't run, it will blow away my machine settings.